Next.jsのバージョンを13.3以降にアップグレードしたい

チャレンジしてみようと思ったらそもそもテストが動かなかった。

```sh
$ docker compose run playwright yarn test
[+] Running 1/0
 ⠿ Container tenkir-app-1  Running                                                                                                                                   0.0s
yarn run v1.22.19
$ yarn playwright test
$ /playwright/node_modules/.bin/playwright test


Error: Cannot find module '@/playwright/models/room-page'
Require stack:
- /playwright/tests/rooms/loadingUntilSocketConnected.spec.ts
- /playwright/node_modules/@playwright/test/lib/common/transform.js
- /playwright/node_modules/@playwright/test/lib/common/testType.js
- /playwright/node_modules/@playwright/test/lib/common/test.js
- /playwright/node_modules/@playwright/test/lib/reporters/multiplexer.js
- /playwright/node_modules/@playwright/test/lib/runner/reporters.js
- /playwright/node_modules/@playwright/test/lib/runner/runner.js
- /playwright/node_modules/@playwright/test/lib/cli.js
- /playwright/node_modules/playwright-core/lib/cli/cli.js
- /playwright/node_modules/playwright-core/cli.js
- /playwright/node_modules/playwright/cli.js

   at rooms/loadingUntilSocketConnected.spec.ts:2

  1 | import { expect, test } from '@playwright/test';
> 2 | import RoomPage from '@/playwright/models/room-page';
    | ^
  3 | import createRoomId from '@/playwright/helpers/createRoomId';
  4 |
  5 | test('ルームを参加するとき、WebSocketが確立すると入室完了トーストが表示されること', async ({

    at Object.<anonymous> (/playwright/tests/rooms/loadingUntilSocketConnected.spec.ts:2:1)


To open last HTML report run:

  npx playwright show-report

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/package.json
+++ b/package.json
@@ -27,7 +27,7 @@
     "@fortawesome/react-fontawesome": "^0.2.0",
     "bufferutil": "^4.0.7",
     "express": "^4.18.2",
-    "next": "13.2.4",
+    "next": "^13.4.13",
     "node-sass": "^9.0.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",

--- a/pages/api/socket.ts
+++ b/pages/api/socket.ts
@@ -1,6 +1,7 @@
 import { NextApiRequest, NextApiResponse } from 'next';
-import { Server as NetServer, Socket } from 'net';
-import { Server as SocketIOServer } from 'socket.io';
+import { Socket } from 'net';
+import { Http2SecureServer } from 'http2';
+import { Server } from 'socket.io';
 
 // class
 import { Room } from '@/class/room';
@@ -17,17 +18,17 @@ import { findRoomById } from './utils/findRoomById';
 
 type NextApiResponseSocketIO = NextApiResponse & {
   socket: Socket & {
-    server: NetServer & {
-      io: SocketIOServer;
+    server: Http2SecureServer & {
+      io: Server;
     };
   };
 };
 
 const SocketHandler = (req: NextApiRequest, res: NextApiResponseSocketIO) => {
   if (!res.socket.server.io) {
-    const io = new SocketIOServer<IFClientToServerEvents, IFServerToClientEvents>(
-      res.socket.server as any,
-    );
+    const io = new Server<IFClientToServerEvents, IFServerToClientEvents>(res.socket.server, {
+      addTrailingSlash: false,
+    });
     const rooms: Room[] = [];
 
     const removeRoomById = (roomId: string): void => {

--- a/playwright/tests/head/ogp.spec.ts
+++ b/playwright/tests/head/ogp.spec.ts
@@ -12,7 +12,7 @@ const description: string =
   'チームでの見積もりをリアルタイムで共有できます。' +
   '参加者はスマートフォンやパソコンのブラウザだけで参加できます。' +
   '直感的なUIと使いやすさが魅力です。';
-const ogImageUrl: RegExp = /opengraph-image\..*\.jpg/;
+const ogImageUrl: RegExp = /opengraph-image\.jpg/;
 
 test('トップページで、OGが正しいこと', async ({ page }) => {
   // Given

